features/moviedetail/src/main/java/com/tkpd/moviedetail/view/MovieDetailFragment.kt

```diff
@@ -47,53 +60,83 @@ class MovieDetailFragment : Fragment() {
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         arguments?.let {
-            movieId = it.getInt(MovieConstant.PARAM_MOVIE_ID, 0)
+            movieId = it.getInt(PARAM_MOVIE_ID, 0)
         }
     }
 
     override fun onCreateView(
         inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?
-    ): View? {
-        // Inflate the layout for this fragment
-        return inflater.inflate(R.layout.fragment_movie_detail, container, false)
+    ): View {
+        return ComposeView(requireContext()).apply {
+            setContent {
+                MovieDetailMainLayout(viewModel)
+            }
+        }
     }
 
+
     override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
         super.onViewCreated(view, savedInstanceState)
-        showLoading()
-        getErrorLayout.hide()
         viewModel.getMovieList(movieId ?: 0)
     }
 
-    override fun onActivityCreated(savedInstanceState: Bundle?) {
-        super.onActivityCreated(savedInstanceState)
-        viewModel.movieDetail.observe(viewLifecycleOwner, Observer { data ->
-            data.doSuccessOrFail({
-                getErrorLayout.hide()
-                renderView(it.data ?: MovieDetail())
-            }, {
-                getErrorLayout.show()
-            }, ::showLoading, ::hideLoading)
-        })
+    @Composable
+    fun MovieDetailMainLayout(viewModel: MovieDetailViewModel = viewModel()) {
```

Comment:
will be good if you can diff file between fragment and layout compose function
1. MovieDetailFragment.kt
    - provide viewmodel, navigation, get args only 
    - call MovieDetailMainLayout() func in point 2
2. MovieDetailScreen.kt or MovieDetailMainLayout.kt
   - MovieDetailMainLayout() func
   - ErrorContent() func
   - ProgressBar() func
   - Preview